java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java

```diff
@@ -291,6 +275,88 @@ public Optional<BulkDataRules> getApplicableRules(RuleSet rules, String sourceOb
         return Optional.ofNullable(applicableRules);
     }
 
+
+    /**
+     * validate that the input stream can be processed per request/transform
+     *
+     * use-case: avoid opening output writer if file isn't valid
+     *
+     * @param request
+     * @param transform
+     * @param inputStreamSupplier
+     *
+     * @return
+     */
+    @SneakyThrows
+    boolean validate(StorageEventRequest request,
+                     StorageHandler.ObjectTransform transform,
+                     Supplier<InputStream> inputStreamSupplier) {
+        int bufferSize = getBufferSize();
+        try (
+            InputStream decompressedStream = request.getDecompressInput() ? new GZIPInputStream(inputStreamSupplier.get(), bufferSize) : inputStreamSupplier.get();
+            BufferedReader reader = new BufferedReader(new InputStreamReader(decompressedStream, StandardCharsets.UTF_8), bufferSize);
+            OutputStream out = new ByteArrayOutputStream()
+        ) {
+
+            StringBuffer firstLines = new StringBuffer();
+            while (firstLines.length() < 5) {
+                String line = reader.readLine();
+                if (line == null) {
+                    break;
+                }
+                firstLines.append(line);
+            }
+
+            Supplier<InputStream> firstLinesSupplier = () -> new ByteArrayInputStream(firstLines.toString().getBytes());
+
+            this.process(request, transform, firstLinesSupplier, () -> out);
+
+            return true;
```

Comment:
The way that `validate` fails is going to be if `process` raise an exception, right? Then, I'd drop the `return true` even not used on the caller; just return nothing and leave the exception flow.